org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/grammarAccess/GrammarAccessExtensions.xtend

```diff
@@ -396,8 +392,49 @@ class GrammarAccessExtensions {
 		}
 	}
 	
-	def String grammarFragmentToString(EObject ele, String prefix) {
-		return serializer.grammarFragmentToString(ele, prefix)
+	def String grammarFragmentToString(EObject object, String prefix) {
+		val node = NodeModelUtils.findActualNodeFor(object)
+		if (node === null) {
+			if (object instanceof RuleCall)
+				if (object?.rule?.name !== null)
+					return process(object.rule.name, prefix)
+			return ""
+		} else {
+			return node.text.process(prefix)
+		}
+	}
+
+	private def process(String input, String prefix) {
+		// remove leading and trailing blank lines
+		var lines = input.split('(\\r?\\n)')
+		var first = 0
+		while (isBlank(lines.get(first)))
+			first++
+		var last = lines.length - 1
+		while (isBlank(lines.get(last)))
+			last--
+		lines = lines.subList(first, last + 1)
+		// remove common whitespace (e.g. leading blanks) and add prefix to each line
+		val commonWhitespace = commonLeadingWhitespace(lines)
```

Comment:
commonLeadingWhitespace ignores blank lines. Let's assume we do have three lines:
- `   text`
- ` `
- `   more text`

The commonWhitespace should be 3.
The second line has only a single char, though. I'd assume that the logic in line 420 will now fail: `lines.get(1).substring(3)` will throw an IOOBE.

I'd say blank lines should be replaced by the prefix itself.

Reply:
Good catch. Will fix.